spellbook/models/balances/ethereum/erc721/balances_ethereum_erc721_hour.sql

```diff
@@ -13,22 +13,18 @@ with
             ) as hour
     )
 
-, daily_balances as
- (SELECT
-    wallet_address,
-    token_address,
-    tokenId,
-    hour,
-    lead(hour, 1, now()) OVER (PARTITION BY token_address, wallet_address ORDER BY hour) AS next_hour
-    FROM {{ ref('transfers_ethereum_erc721_rolling_hour') }})
-
-SELECT distinct
+SELECT
     'ethereum' as blockchain,
     d.hour,
     b.wallet_address,
     b.token_address,
     b.tokenId,
-    nft_tokens.name as collection
-FROM daily_balances b
-INNER JOIN hours d ON b.hour <= d.hour AND d.hour < b.next_hour
-LEFT JOIN {{ ref('tokens_ethereum_nft') }} nft_tokens ON nft_tokens.contract_address = b.token_address
+    b.num_tokens,
+    nft_tokens.name as collection,
+    nft_tokens.category
+FROM hours d
+INNER JOIN {{ ref('transfers_ethereum_erc721_rolling') }} b ON (b.evt_block_time <= d.hour AND d.hour < b.next_evt)
```

Comment:
same as other comment

minor feedback: for readability, even though with inner join the results are the same, can we switch the FROM to be on {{ ref('transfers_ethereum_erc721_rolling') }} and the JOIN on hours? as an outsider looking in for first time, seems that would flow better